freelist.go

```diff
@@ -160,8 +277,98 @@ func (f *freelist) release(txid txid) {
 			delete(f.pending, tid)
 		}
 	}
-	sort.Sort(m)
-	f.ids = pgids(f.ids).merge(m)
+
+	for _, pgid := range m {
+		// try to see if we can merge and update
+		f.mergeWithExistingSpan(pgid)
+	}
+}
+
+// mergeWithExistingSpan merges pgi to the existing free spans, try to merge it backward and forward
+func (f *freelist) mergeWithExistingSpan(pgi pgid) {
```

Comment:
this function can be written like:
start, size, pgi := ...
if prev {
    remove prev span
    start, size, pgi := ...
}
if next {
    remove next span
    start, size, pgi := ...
}
add this span ...

Reply:
@hormes pal

Reply:
pedal

Reply:
pedal